deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -647,3 +647,189 @@ func promoteAllCandidatesForChainOps(
 
 	return mcmsOps, nil
 }
+
+type RevokeCandidateChangesetConfig struct {
+	HomeChainSelector uint64
+
+	// DONChainSelector is the chain selector whose candidate config we want to revoke.
+	DONChainSelector uint64
+	NodeIDs          []string
+	PluginType       types.PluginType
+
+	// MCMS is optional MCMS configuration, if provided the changeset will generate an MCMS proposal.
+	// If nil, the changeset will execute the commands directly using the deployer key
+	// of the provided environment.
+	MCMS *MCMSConfig
+}
+
+func (r RevokeCandidateChangesetConfig) Validate(e deployment.Environment, state CCIPOnChainState) (deployment.Nodes, error) {
```

Comment:
if we at all put the `validate` as part of interface (to ensure `Validate` is called on all configs), the standardization of method signature would be difficult. Can we get nodes as part of a diff method?

Reply:
I think it'd be tough to have it as part of the interface tbh, the family of validation funcs is already all over the place.